File: services/portfolio-analytics/src/core/domain/entities/trading.py
from __future__ import annotations

from dataclasses import dataclass
from datetime import datetime
from uuid import UUID


@dataclass
class Order:
    order_id: UUID
    account_id: UUID
    instrument_id: UUID
    side: str
    quantity: float
    order_type: str
    status: str
    created_at: datetime


@dataclass
class Trade:
    trade_id: UUID
    order_id: UUID
    quantity: float
    price: float
    fee: float | None
    executed_at: datetime

File: apps/server-go/internal/domain/trading.go
package domain

import "time"

type Order struct {
	OrderID     string    `json:"order_id"`
	AccountID   string    `json:"account_id"`
	InstrumentID string   `json:"instrument_id"`
	Side        string    `json:"side"`
	Quantity    float64   `json:"quantity"`
	OrderType   string    `json:"order_type"`
	Status      string    `json:"status"`
	CreatedAt   time.Time `json:"created_at"`
}

type Trade struct {
	TradeID    string    `json:"trade_id"`
	OrderID    string    `json:"order_id"`
	Quantity   float64   `json:"quantity"`
	Price      float64   `json:"price"`
	Fee        *float64  `json:"fee,omitempty"`
	ExecutedAt time.Time `json:"executed_at"`
}
